Address Safer CPP warnings in HTMLDialogElement.cpp

nt1m opened the pull request:
#### 6c9b009a367989e7cb150d128bd9b5f70bb46abd
<pre>
Address Safer CPP warnings in HTMLDialogElement.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=289224">https://bugs.webkit.org/show_bug.cgi?id=289224</a>
<a href="https://rdar.apple.com/146363685">rdar://146363685</a>

Reviewed by Ryosuke Niwa.

* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLambdaCapturesCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/html/HTMLDialogElement.cpp:
(WebCore::HTMLDialogElement::show):
(WebCore::HTMLDialogElement::showModal):
(WebCore::HTMLDialogElement::queueCancelTask):
(WebCore::HTMLDialogElement::runFocusingSteps):
(WebCore::HTMLDialogElement::queueDialogToggleEventTask):

Canonical link: <a href="https://commits.webkit.org/291693@main">https://commits.webkit.org/291693@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5510b39da14b1cc6fc964d0855f483847153fd23

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/93704 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/13277 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/3025 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/98710 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/44230 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/95754 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/13572 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/21720 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/71536 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/28907 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/96706 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/10103 "Found 1 new test failure: fast/forms/ios/focus-input-via-button-no-scaling.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/84700 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/51870 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/9785 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/2327 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/43545 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/80058 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/2395 "Passed tests") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/100769 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/20756 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/15142 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/80551 "layout-tests (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/21008 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/80642 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/79889 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/24436 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/1798 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/13909 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/15034 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/20740 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/25918 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/20427 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/23887 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/22168 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/html/HTMLDialogElement.cpp:
```diff
@@ -77,11 +77,11 @@ ExceptionOr<void> HTMLDialogElement::show()
     setBooleanAttribute(openAttr, true);
 
-    m_previouslyFocusedElement = document().focusedElement();
+    m_previouslyFocusedElement = protectedDocument()->focusedElement();
```
Instead of calling protectedDocument() twice in row.
Can we allocate a `RefPtr` local variable and store document there?

rniwa commented on Source/WebCore/html/HTMLDialogElement.cpp:
```diff
@@ -139,11 +139,11 @@ ExceptionOr<void> HTMLDialogElement::showModal()
     RenderElement::markRendererDirtyAfterTopLayerChange(this->checkedRenderer().get(), containingBlockBeforeStyleResolution.get());
 
-    m_previouslyFocusedElement = document().focusedElement();
+    m_previouslyFocusedElement = protectedDocument()->focusedElement();
```
Ditto.

nt1m committed:
Address Safer CPP warnings in HTMLDialogElement.cpp
https://bugs.webkit.org/show_bug.cgi?id=289224
rdar://146363685

Reviewed by Ryosuke Niwa.

* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLambdaCapturesCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/html/HTMLDialogElement.cpp:
(WebCore::HTMLDialogElement::show):
(WebCore::HTMLDialogElement::showModal):
(WebCore::HTMLDialogElement::queueCancelTask):
(WebCore::HTMLDialogElement::runFocusingSteps):
(WebCore::HTMLDialogElement::queueDialogToggleEventTask):

Canonical link: https://commits.webkit.org/291693@main

cdumez commented on Source/WebCore/html/HTMLDialogElement.cpp:
```diff
@@ -104,7 +105,8 @@ ExceptionOr<void> HTMLDialogElement::showModal()
         return Exception { ExceptionCode::InvalidStateError, "Element is already an open popover."_s };
 
-    if (!protectedDocument()->isFullyActive())
+    Ref document = this->document();
```
Please revert his change, this is a step in the wrong direction.

cdumez replied:
Disregard this comment, I thought this variable was only used once on the next line. But it is used later in this function, a local variable is the right thing to do in this case. Reading diffs is hard sometimes :P
